src/nvs.rs

```diff
@@ -810,28 +814,187 @@ impl RawHandle for EspNvs<NvsEncrypted> {
     }
 }
 
-impl<T: NvsPartitionId> StorageBase for EspNvs<T> {
+impl<T: NvsPartitionId> StorageBase for EspKeyValueStorage<T> {
     type Error = EspError;
 
     fn contains(&self, name: &str) -> Result<bool, Self::Error> {
-        EspNvs::contains(self, name)
+        EspKeyValueStorage::contains(self, name)
     }
 
     fn remove(&mut self, name: &str) -> Result<bool, Self::Error> {
-        EspNvs::remove(self, name)
+        EspKeyValueStorage::remove(self, name)
     }
 }
 
-impl<T: NvsPartitionId> RawStorage for EspNvs<T> {
+impl<T: NvsPartitionId> RawStorage for EspKeyValueStorage<T> {
     fn len(&self, name: &str) -> Result<Option<usize>, Self::Error> {
-        EspNvs::len(self, name)
+        EspKeyValueStorage::len(self, name)
     }
 
     fn get_raw<'a>(&self, name: &str, buf: &'a mut [u8]) -> Result<Option<&'a [u8]>, Self::Error> {
-        EspNvs::get_raw(self, name, buf)
+        EspKeyValueStorage::get_raw(self, name, buf)
     }
 
     fn set_raw(&mut self, name: &str, buf: &[u8]) -> Result<bool, Self::Error> {
-        EspNvs::set_raw(self, name, buf)
+        EspKeyValueStorage::set_raw(self, name, buf)
+    }
+}
+
+impl<T: NvsPartitionId> EspKeyValueStorage<T> {
+    pub fn new(
```

Comment:
Perhaps just provide instead a `const fn new(nvs: EspNvs)` instead?

Reply:
Also since the `EspKeyValueStorage` "owns" the `EspNvs` instance, perhaps provide an `pub const fn esp_nvs(&self) -> &EspNvs {}` method.